python/tests/stress/arcticdb/test_mem_leaks.py

```diff
@@ -0,0 +1,201 @@
+        print (f"  Overall stats : Process growth since start {nice_bytes_str(process_growth)} AVG growth per iter {nice_bytes_str(process__avg_growth)}")
+        print (f"  Minimum growth so far: {nice_bytes_str(min(mem_growth_each_iter))}")
+        print (f"  Maximum growth so far: {nice_bytes_str(max(mem_growth_each_iter))}")
+        print (f"  Number of times there was 50% drop in memory: {count_drops(mem_growth_each_iter, 0.5)}")
+
+        assert max_total_mem_lost_treshold >= process_growth , f"Memory of the process grew more than defined threshold: {nice_bytes_str(process_growth)} (specified: {nice_bytes_str(max_total_mem_lost_treshold)} )"
+        assert max_machine_memory_percentage >= mem_per , f"Machine utilized more memory than specified treshold :{mem_per}% (specified {max_machine_memory_percentage}%)"
+        
+    print ("The process assesment finished within expectations. Total consumed additionaml mem is bellow threshold: ", process_growth)
+    print ("Returning this number as result.")
+    return process_growth, count_drops(mem_growth_each_iter, 0.5)
+
+def grow_exp(df_to_grow : pd.DataFrame, num_times_xx2:int):
+    """
+        Quickly inflates the dataframe by doubling its size for each iteration
+        Thus the final size will be 'num_times_xx2' power of two
+    """
+    for n in range(1,num_times_xx2):
+        df_prev = df_to_grow.copy(deep=True)
+        df_to_grow = pd.concat([df_to_grow, df_prev])
+    return df_to_grow
+
+@pytest.mark.xfail(reason = "ArcticDBPR#1998 String memory leaks while read")
+def test_mem_leak_read_all_arctic_lib(arctic_library_lmdb):
+    lib : adb.Library = arctic_library_lmdb
+
+    df = get_sample_dataframe(size=1000000)
+    df = grow_exp(df, 5)
+
+    symbol = "test"
+    lib.write(symbol, df)
+
+    def proc_to_examine():
+        data = lib.read(symbol).data
+        del data
+
+    df = lib.read(symbol).data
+    print(df)
+    del df
+
+    '''
+        To determine optimal max_mem first set the procedure to a very high number like 10 GB
+        and run it from commandline for approx 10 iteration.
+
+        > python -m pytest -s python/tests/stress/arcticdb/test_mem_leaks.py::test_mem_leak_read_all_arctic_lib
+
+        Observe 2 things:
+         - maximum process memory growth
+         - ideally there should be one iteration the mem did go down indicating GC in C++ code
+
+         if both conditions are met then the process is likely to not leak big chunks of mem
+         and you can take the max mem growth number add 20 percent safety margin and you will
+         arrive at reasonable  max_mem
+
+         Then put for number of iterations 3x or 5x and this mem limit and 
+         run the test from command line again to assure it runs ok before commit 
+
+    '''
+    max_mem = 4808662528
```

Comment:
this is bytes ...

Now scaling down is indeed possible, and perhaps I will do that now when things are fixed. Before I wanted to give the process maximum possible iterrations (7) on my machine to survive

Reply:
`153623040` is still about 153GB isn't it? Surely we don't expect this to use so much memory?

Reply:
I have started the execution couple of times it was 116MB the first the the max and less than 100MB second execution This means that we cannot set fix number, therefore I assigned a litle bit of safety margin. As we execute it many times the idea is that it would stay lower that number. If we try to optimize it I think we could end in a situation where the test becaomes flaky. Sometimes it will go above that number and when you run it all will be well the next time. Thus I recommend staying that way

Reply:
I did the units conversion wrong. This looks good.